### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by Gdiazdiaz**
## Tasks completed

- Add classroom, books, and rental class
- Create relationships between

1. Classroom and Student

2. Classroom and Student

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  attr_accessor :label
+
+  def add_student(student)
+    @students << film
```
Comment (m05e5) [inline classroom.rb]: - [ ] Hi you did a great job but you're supposed to append the student variable to your @students instance variable 
`@students << student`

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  attr_reader :rentals
+
+  def initialize(author:, title:)
+    @author = author
+    @title = title
+    @rentals = []
+  end
+
+  attr_accessor :author, :tittle
+end
```
Comment (m05e5) [inline book.rb]: - [ ] As part of the requirements, you are supposed to create a has-many relationship side of Book and Rental. So kindly add a method to add rentals in the book Class

File: person.rb [hunk 3]
```diff
@@ -8,6 +10,7 @@ def initialize(id:, age:, name: 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
```
Comment (m05e5) [inline person.rb:13]: - [ ]  As part of the requirements, you are supposed to create a has-many relationship side of Person and Rental. So kindly add a method to add rentals in the person Class

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @Gdiazdiaz  , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights

- Correct gitflow
- No linters error
- Clean and organized code

## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] Please check the comments below 

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    @title = title
+    @rentals = []
+  end
+
+  attr_accessor :author, :tittle
+
+  def add_rental(rental)
+    @rentals << rental
+    rental.book = self
+  end
```
Comment (Mihndim2020) [inline book.rb]: - [ ] As requested by the previous code reviewer, which is a requirement of this project, The `add_rental` method takes `person`  and `date` as  parameters, and `book` itself is sent as a parameter to create the new rental achieving the 'has-many' association. Something like so: 

```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

Please kindly fix that.

File: person.rb [hunk 5]
```diff
@@ -37,6 +40,11 @@ def correct_name
     @name
   end
 
+  def add_rental(rental)
+    @rentals << rental
+    rental.person = self
+  end
+
   private
```
Comment (Mihndim2020) [inline person.rb]: - [ ] As requested by the previous code reviewer, which is a requirement for this project, The `add_rental` method takes `book`  and `date` as  parameters, and  `person` itself is sent as a parameter to create the new rental achieving the 'has-many' association. Something like so: 

```ruby
  def add_rental(book, date)
    Rental.new(date, self, book)
  end
```

Please kindly fix that.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
## Hi @Gdiazdiaz, 

Great work on making the changes requested by a previous reviewer 👏 👏 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
stay-strong-youre-almost-there


 # Status: Requires Changes ♻️
_Check the comments under the review._


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Gdiazdiaz, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Correct `GitFlow` ✔️ 
- Good implementations ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mukhammadrizooff** [comment]
@Gdiazdiaz Good luck 😃

**Pull request merged**